subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c

```diff
@@ -522,6 +522,18 @@ static int dc_rx_data_handler(const struct nct_evt *nct_evt)
 		.topic = nct_evt->param.dc->topic,
 	};
 
+	if (nrf_cloud_detect_disconnection_request(nct_evt->param.dc->data.ptr)) {
+		LOG_DBG("nRF Cloud has requested us to disconnect, "
+			"likely due to device deletion");
```

Comment:
you could update this to say with certainty that the device has been removed from nrf cloud account since it appears to be the only reason this message is sent.